analyzers/tests/SonarAnalyzer.Test/Trackers/ArgumentTrackerTest.cs

```diff
@@ -0,0 +1,1288 @@
+    [DataRow("""<AttributeUsage(AttributeTargets.All,  $$AllowMultiple := true)>""", "AllowMultiple", true)]
+    [DataRow("""<AttributeUsage(AttributeTargets.All,  $$AllowMultiple := true, Inherited := true)>""", "AllowMultiple", true)]
+    [DataRow("""<AttributeUsage(AttributeTargets.All,  $$AllowMultiple := true, Inherited := true)>""", "Inherited", false)]
+    [DataRow("""<AttributeUsage(AttributeTargets.All,  AllowMultiple := true, $$Inherited := true)>""", "Inherited", true)]
+    public void Attribute_Property_VB(string attribute, string propertyName, bool expected)
+    {
+        var snippet = $$"""
+            Imports System
+
+            {{attribute}}
+            Public NotInheritable Class TestAttribute
+                Inherits Attribute
+            End Class
+            """;
+        var (node, model) = ArgumentAndModelVB(snippet);
+
+        var argument = ArgumentDescriptor.AttributeProperty("AttributeUsage", propertyName);
+        var (result, context) = MatchArgumentVB(model, node, argument);
+        result.Should().Be(expected);
+        if (result)
+        {
+            // The mapped parameter is the "value" parameter of the property set method.
+            context.Parameter.Name.Should().Be("value");
+            var method = context.Parameter.ContainingSymbol.Should().BeAssignableTo<IMethodSymbol>().Which;
+            method.MethodKind.Should().Be(MethodKind.PropertySet);
+            method.AssociatedSymbol.Should().BeAssignableTo<IPropertySymbol>().Which.Name.Should().Be(propertyName);
+        }
+    }
+
+    private static string WrapInMethodCS(string snippet) =>
+        $$"""
+        using System;
+        class C
+        {
+            public void M()
+            {
+                {{snippet}}
+            }
+        }
+        """;
+
+    private static string WrapInMethodVB(string snippet) =>
+        $$"""
+        Imports System
+        Class C
+            Public Sub M()
+                {{snippet}}
+            End Sub
+        End Class
+        """;
+
+    private static (SyntaxNode Node, SemanticModel Model) ArgumentAndModel(string snippet,
+        Func<string, MetadataReference[], (SyntaxTree Tree, SemanticModel Model)> compile, params Type[] argumentNodeTypes)
+    {
+        var pos = snippet.IndexOf("$$");
+        if (pos == -1)
+        {
+            throw new InvalidOperationException("The $$ maker was not found");
+        }
+        snippet = snippet.Replace("$$", string.Empty);
+        var (tree, model) = compile(snippet, MetadataReferenceFacade.SystemCollections.Concat(MetadataReferenceFacade.SystemDiagnosticsProcess).ToArray());
+        var node = tree.GetRoot().DescendantNodesAndSelf(new TextSpan(pos, 1)).Reverse().First(x => argumentNodeTypes.Any(t => t.IsInstanceOfType(x))); // root.Find does not work with OmittedArgument
+        return (node, model);
+    }
+
+    private static (SyntaxNode Node, SemanticModel Model) ArgumentAndModelCS(string snippet) =>
```

Comment:
We have a class that stores this information already `(SyntaxNode Node, SemanticModel Model)` -> `ArgumentContext`. I think we should use it instead of the tuple.